### PR TITLE
allow api requests from ngrok-tunnelled websites in development

### PR DESCRIPTION
**Pull request by acburdine**
no issue
- adds a check to middleware/auth that allows requests to the ghost api from ngrok tunnels (while in development mode). This makes it easier for testing.

### CODE REVIEW

**Comment by halfdan** [comment]
This seems oddly specific. What if I want to use localtunnel or any other tunneling solution?

**Comment by acburdine** [comment]
maybe something in config to allow for tunneling?

**Comment by jaswilli** [comment]
Yeah, I don't see why this belongs in the codebase. :-1:

**Comment by acburdine** [comment]
It was based on a comment that @ErisDS made in Slack about ngrok -> however I see your point. I still think it would be nice to have a way to support tunneling in development without having to change the config url each time, but there's probably a better way to do this.

**Comment by ErisDS** [comment]
IMO we need a solution such that we don't have to keep fudging urls when trying to test & develop the software on mobile. I think a single line of code is worth it in this case, unless there is another easy solution?

**Comment by jaswilli** [comment]
What works for me is creating diffs or stashing these local changes and then when they're needed it's as simple as applying them with `patch` or `git stash`.

**Comment by acburdine** [comment]
@jaswilli the problem with that solution (unless you're talking about stashing the changes in this PR) is that unless you pay for ngrok, the url changes each time and thus the stash in invalid.

**Comment by ErisDS** [comment]
This PR has touched on an important issue. I feel like we constantly make developing Ghost _ridiculously_ hard on ourselves. 

I'm not blind to the fact that the proposed fix is a bit dirty, or that in the past this would have been the kind of thing I'd probably say no too as well, but at some point the success of the project itself has to be put ahead of everything else. I want us to be comfortable making our own lives easier sometimes. 

A problem has been identified with the developer workflow for testing on mobile, and I think it is important that those sorts of frictions be identified and eliminated. 

So what else could we do to make it so that testing Ghost on mobile devices is frictionless?

**Comment by acburdine** [comment]
I wonder if we could just have some sort of a flag that could disable the origin checking, but only in development? That would allow for any tunneling request from any service to pass through.

**Comment by jaswilli** [comment]
Adding an open-ended regex into a piece of authentication code to handle a very small edge case of developer workflow seems to be well on the way to paving the road to hell... Especially since the workaround is just updating `config.js`.

That said, I'm totally open to solving this problem--I just don't feel good about the `ngrok` regex. I think this can be solved by taking a look at what this code path in `auth.js` is intended to do--the way I see it, it's supposed to be validating the client credentials and setting CORS headers.

The part of the code where it ruthlessly checks the `origin` header, even for non-CORS requests (reverse-proxying via ngrok is not cross-origin) seems extreme to me. The only thing it's doing is attempting to prevent the API from being proxied, which isn't really practical (i.e., I can proxy to Ghost and tell it the origin is anything I want and then add CORS headers on the way out).

tl;dr: I propose we just use `origin` to make decisions about CORS.

**Comment by jaswilli** [comment]
For those that care about using ngrok, a side-effect of #6644 is that it should work now without any hoop-jumping.

**Comment by ErisDS** [comment]
:+1: :boom:
